src/components/scenes/RampSelectOptionScene.tsx

```diff
@@ -0,0 +1,389 @@
+import { useSelector } from '../../types/reactRedux'
+import type { BuySellTabSceneProps } from '../../types/routerTypes'
+import { getPaymentTypeIcon } from '../../util/paymentTypeIcons'
+import { getPaymentTypeDisplayName } from '../../util/paymentTypeUtils'
+import { AlertCardUi4 } from '../cards/AlertCard'
+import { PaymentOptionCard } from '../cards/PaymentOptionCard'
+import { EdgeAnim } from '../common/EdgeAnim'
+import { SceneWrapper } from '../common/SceneWrapper'
+import { SectionHeader } from '../common/SectionHeader'
+import { styled } from '../hoc/styled'
+import { SceneContainer } from '../layout/SceneContainer'
+import { CardListModal } from '../modals/CardListModal'
+import { ShimmerCard } from '../progress-indicators/ShimmerCard'
+import { Airship } from '../services/AirshipInstance'
+import { useTheme } from '../services/ThemeContext'
+import { EdgeText } from '../themed/EdgeText'
+
+export interface RampSelectOptionParams {
+  rampQuoteRequest: RampQuoteRequest
+}
+
+interface Props extends BuySellTabSceneProps<'rampSelectOption'> {}
+
+export const RampSelectOptionScene: React.FC<Props> = (props: Props) => {
+  const { route } = props
+  const { rampQuoteRequest } = route.params
+
+  const theme = useTheme()
+  const account = useSelector(state => state.core.account)
+  // Get ramp plugins
+  const { data: rampPluginArray = [], isLoading: isPluginsLoading } =
+    useRampPlugins({ account })
+  const rampPlugins = React.useMemo(() => {
+    const map: Record<string, RampPlugin> = {}
+    for (const plugin of rampPluginArray) {
+      map[plugin.pluginId] = plugin
+    }
+    return map
+  }, [rampPluginArray])
+
+  // Use supported plugins hook
+  const { supportedPlugins } = useSupportedPlugins({
+    selectedWallet: rampQuoteRequest.wallet,
+    selectedCrypto:
+      rampQuoteRequest.wallet != null
+        ? {
+            pluginId: rampQuoteRequest.pluginId,
+            tokenId: rampQuoteRequest.tokenId
+          }
+        : undefined,
+    selectedFiatCurrencyCode: rampQuoteRequest.fiatCurrencyCode.replace(
+      'iso:',
+      ''
+    ),
+    countryCode: rampQuoteRequest.regionCode?.countryCode,
+    stateProvinceCode: rampQuoteRequest.regionCode?.stateProvinceCode,
+    plugins: rampPlugins,
+    direction: rampQuoteRequest.direction
+  })
+
+  // Use supported plugins
+  const pluginsToUse = Object.fromEntries(
+    supportedPlugins.map(result => [result.plugin.pluginId, result.plugin])
+  )
+
+  // Use the ramp quotes hook
+  const {
+    quotes: allQuotes,
+    isLoading: isLoadingQuotes,
+    isFetching: isFetchingQuotes,
+    errors: failedQuotes
+  } = useRampQuotes({
+    rampQuoteRequest,
+    plugins: pluginsToUse
+  })
+
+  const handleQuotePress = async (quote: RampQuoteResult): Promise<void> => {
+    try {
+      await quote.approveQuote({
+        coreWallet: rampQuoteRequest.wallet!
+      })
+    } catch (error) {
+      console.error('Failed to approve quote:', error)
+    }
+  }
```

Comment:
Should `showError`. Also the non-null assertion wallet! is safe given your creation flow, but this scene’s param type makes wallet optional. Consider guarding, or make wallet required in the route params to avoid future regressions.

Reply:
This try-catch will be removed. We're not suppose to handle these unexpected errors here.